test/strings.jl

```diff
@@ -742,6 +742,10 @@ u = SubString(str, 1, 5)
 @test rsearch(u, 'z') == 0
 @test rsearch(u, "ll") == 3:4
 
+# rsearch with a two utf8-characters query (issue #9365)
+@test rsearch("ééé", "éé") == 3:5
+@test rsearchindex("ééé", "éé", endof("ééé")) == 3
```

Comment:
Could you also add tests with UTF-8 characters with more than two bytes, to make sure we never rely on this assumption? `€` is a good example.